repocreation/createRepo.go

```diff
@@ -0,0 +1,212 @@
+package repocreation
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/google/go-github/v28/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"os"
+	"strings"
+)
+
+const (
+	org = "ONSdigital"
+	// Team: "DigitalPublishing", slug: "digitalpublishing", id: 779417
+	dpTeamSlug = "DigitalPublishing"
+	teamID = int64(779417)
+	defaultBranch = "develop"
+	masterBranch = "master"
+)
+
+// GenerateGithubRepository is the entry point to generating the repository
+func GenerateGithubRepository() error {
+	fmt.Println("This script will create a new ONS Digital Publishing repository." +
+		"In order to create and configure a new repository please answer the prompts.")
+
+	accessToken, userHandle, repoName, repoDescription := getConfigurationsForNewRepo()
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	checkAccess(ctx, client)
+
+	repo := &github.Repository{
+		Name:          github.String(repoName),
+		Description:   github.String(repoDescription),
+		DefaultBranch: github.String(defaultBranch),
+		MasterBranch:  github.String(masterBranch),
+		Private:       github.Bool(false),
```

Comment:
Would it be useful to allow the user to decide whether the repository should be private or public?

Possibly handle an optional flag when running script? Defaults to false if not set

Reply:
I think we have hit our organisation limit of private repos, so the user wouldn't be able to create a private one with this tool unfortunatly.